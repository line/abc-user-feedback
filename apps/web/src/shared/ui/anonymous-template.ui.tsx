/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import Image from 'next/image';

import { Icon } from '@ufb/react';

import { useTenantStore } from '@/entities/tenant';

import { cn } from '../utils';

interface Props extends React.PropsWithChildren {
  title: string;
  image: string;
  imageSub?: React.ReactNode;
  description?: string;
}

const AnonymousTemplate = (props: Props) => {
  const { title, children, image, imageSub, description } = props;
  const { tenant } = useTenantStore();

  return (
    <div className="flex h-full items-center justify-center">
      <div className="flex items-stretch justify-center gap-4">
        <div className="border-neutral-tertiary rounded-24 shadow-default flex w-[520px] flex-col gap-10 border p-8">
          <div className="flex h-[54px] justify-between">
            <div>
              <h3 className="text-xlarge-strong">{title}</h3>
              {description && (
                <p className="text-small-normal text-neutral-tertiary whitespace-pre-line">
                  {description}
                </p>
              )}
            </div>
            {tenant && (
              <span>
                <Icon name="RiBuildingLine" size={16} />
                <span className="ml-1">{tenant.siteName}</span>
              </span>
            )}
          </div>
          {children}
        </div>
        <div
          className={cn('flex w-[520px] flex-col items-center justify-end', {
            'justify-center': !!imageSub,
          })}
        >
          <Image src={image} alt="sign-in-image" width={240} height={240} />
          {imageSub}
        </div>
      </div>
    </div>
  );
};

export default AnonymousTemplate;
