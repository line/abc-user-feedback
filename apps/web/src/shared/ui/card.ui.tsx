/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';

import { cn } from '../utils';

type Size = 'sm' | 'md' | 'lg';
const CardContext = React.createContext<{ size: Size }>({
  size: 'md',
});
interface Props
  extends React.PropsWithChildren,
    React.HTMLAttributes<HTMLDivElement> {
  size?: Size;
  className?: string;
}

export const Card = (props: Props) => {
  const { size = 'md', children, className, ...otherProps } = props;

  return (
    <CardContext.Provider value={{ size }}>
      <div
        className={cn(
          'rounded-8 border-neutral-tertiary bg-neutral-primary flex flex-col gap-4 border py-6', // shadow-default
          className,
          { 'gap-4 py-4': size === 'sm' },
        )}
        {...otherProps}
      >
        {children}
      </div>
    </CardContext.Provider>
  );
};

export const CardBody = (
  props: React.PropsWithChildren & React.HTMLAttributes<HTMLDivElement>,
) => {
  const { children, className, ...divProps } = props;

  const { size } = React.useContext(CardContext);

  return (
    <div
      className={cn('px-6', { 'px-4': size === 'sm' }, className)}
      {...divProps}
    >
      {children}
    </div>
  );
};

export const CardHeader = (
  props: React.PropsWithChildren &
    React.HTMLAttributes<HTMLDivElement> & {
      action?: React.ReactNode;
    },
) => {
  const { children, className, action, ...divProps } = props;

  const { size } = React.useContext(CardContext);

  return (
    <div
      className={cn(
        'flex items-center justify-between gap-6 px-6',
        { 'px-4': size === 'sm' },
        className,
      )}
      {...divProps}
    >
      <div>{children}</div>
      {action}
    </div>
  );
};

export const CardTitle = (
  props: React.PropsWithChildren &
    React.HTMLAttributes<HTMLHeadingElement> & { size?: Size },
) => {
  const { children, className, ...divProps } = props;

  const { size } = React.useContext(CardContext);

  return (
    <h4
      className={cn(
        {
          'text-title-h4': (props.size ?? size) === 'lg',
          'text-title-h5': (props.size ?? size) === 'md',
          'text-base-strong': (props.size ?? size) === 'sm',
        },
        className,
      )}
      {...divProps}
    >
      {children}
    </h4>
  );
};

export const CardDescription = (
  props: React.PropsWithChildren &
    React.HTMLAttributes<HTMLHeadingElement> & {},
) => {
  const { children, className, ...divProps } = props;

  return (
    <p
      className={cn('text-neutral-tertiary text-small-normal', className)}
      {...divProps}
    >
      {children}
    </p>
  );
};
