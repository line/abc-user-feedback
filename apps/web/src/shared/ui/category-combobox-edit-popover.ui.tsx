/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { useState } from 'react';
import { useMutation, useQueryClient } from '@tanstack/react-query';
import { useTranslation } from 'react-i18next';

import {
  Badge,
  Dropdown,
  DropdownContent,
  DropdownItem,
  DropdownTrigger,
  Icon,
  TextInput,
  toast,
} from '@ufb/react';

import type { Category } from '@/entities/category';

import { client } from '../lib';

interface Props {
  projectId: number;
  cateogry: Category;
}

const CategoryComboboxEditPopover = (props: Props) => {
  const { projectId, cateogry } = props;

  const { t } = useTranslation();
  const queryClient = useQueryClient();

  const refetch = async () => {
    await queryClient.invalidateQueries({
      queryKey: ['/api/admin/projects/{projectId}/categories'],
    });
    await queryClient.invalidateQueries({
      queryKey: ['/api/admin/projects/{projectId}/issues/search'],
    });
  };

  const [inputValue, setInputValue] = useState(cateogry.name);

  const { mutate: updateCategory } = useMutation({
    mutationFn: async ({ name }: { name: string }) => {
      const { data } = await client.put({
        path: '/api/admin/projects/{projectId}/categories/{categoryId}',
        pathParams: { projectId, categoryId: cateogry.id },
        body: { name },
      });
      return data;
    },
    async onSuccess() {
      await refetch();
      toast.success(t('v2.toast.success'));
    },
    onError(error) {
      toast.error(error.message);
    },
  });
  const { mutate: deleteCategory } = useMutation({
    mutationFn: async () => {
      const { data } = await client.delete({
        path: '/api/admin/projects/{projectId}/categories/{categoryId}',
        pathParams: { projectId, categoryId: cateogry.id },
      });
      return data;
    },
    async onSuccess() {
      await refetch();
      toast.success(t('v2.toast.success'));
    },
    onError(error) {
      toast.error(error.message);
    },
  });
  return (
    <Dropdown>
      <DropdownTrigger asChild>
        <Badge variant="subtle" onClick={(e) => e.stopPropagation()}>
          Edit
        </Badge>
      </DropdownTrigger>
      <DropdownContent onClick={(e) => e.stopPropagation()} side="right">
        <div className="border-b">
          <TextInput
            className="border-none"
            value={inputValue}
            onChange={(e) => setInputValue(e.currentTarget.value)}
            onKeyDown={(e) => {
              if (e.key === 'Enter') {
                e.stopPropagation();
                updateCategory({ name: inputValue });
              }
            }}
          />
        </div>
        <DropdownItem
          className="gap-2"
          onSelect={() => deleteCategory(undefined)}
        >
          <Icon name="RiDeleteBin5Fill" size={16} />
          Delete Category
        </DropdownItem>
      </DropdownContent>
    </Dropdown>
  );
};

export default CategoryComboboxEditPopover;
