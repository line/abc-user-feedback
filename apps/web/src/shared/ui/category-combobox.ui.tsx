/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import React, { useState } from 'react';
import { useRouter } from 'next/router';
import { useMutation, useQueryClient } from '@tanstack/react-query';
import { useTranslation } from 'next-i18next';
import { useThrottle } from 'react-use';

import {
  Combobox,
  ComboboxContent,
  ComboboxGroup,
  ComboboxInput,
  ComboboxItem,
  ComboboxList,
  ComboboxTrigger,
  toast,
} from '@ufb/react';

import type { Category } from '@/entities/category';

import { client, useOAIQuery, usePermissions } from '../lib';
import { cn } from '../utils';
import CategoryComboboxEditPopover from './category-combobox-edit-popover.ui';
import InfiniteScrollArea from './infinite-scroll-area.ui';

interface Props extends React.PropsWithChildren {
  category?: Category | null;
  issueId: number;
}
const LIMIT = 20;
const CategoryCombobox = (props: Props) => {
  const { category, issueId, children } = props;

  const perms = usePermissions();
  const router = useRouter();
  const projectId = Number(router.query.projectId);
  const { t } = useTranslation();

  const [inputValue, setInputValue] = useState('');
  const throttledvalue = useThrottle(inputValue, 500);
  const [page, setPage] = useState(1);

  const queryClient = useQueryClient();

  const { data, isLoading } = useOAIQuery({
    path: '/api/admin/projects/{projectId}/categories',
    variables: { projectId, categoryName: throttledvalue, limit: page * LIMIT },
  });

  const refetch = async () => {
    await queryClient.invalidateQueries({
      queryKey: ['/api/admin/projects/{projectId}/categories'],
    });
    await queryClient.invalidateQueries({
      queryKey: ['/api/admin/projects/{projectId}/issues/search'],
    });
  };

  const { mutate: attachCategory } = useMutation({
    mutationFn: async ({ categoryId }: { categoryId: number }) => {
      const { data } = await client.put({
        path: '/api/admin/projects/{projectId}/issues/{issueId}/category/{categoryId}',
        pathParams: { projectId, issueId, categoryId },
      });
      return data;
    },
    async onSuccess() {
      await refetch();
      toast.success(t('v2.toast.success'));
    },
    onError(error) {
      toast.error(error.message);
    },
  });
  const { mutate: detachCategory } = useMutation({
    mutationFn: async ({ categoryId }: { categoryId: number }) => {
      const { data } = await client.delete({
        path: '/api/admin/projects/{projectId}/issues/{issueId}/category/{categoryId}',
        pathParams: { projectId, issueId, categoryId },
      });
      return data;
    },
    async onSuccess() {
      await refetch();
      toast.success(t('v2.toast.success'));
    },
    onError(error) {
      toast.error(error.message);
    },
  });

  const { mutateAsync: createCategory } = useMutation({
    mutationFn: async ({ name }: { name: string }) => {
      const { data } = await client.post({
        path: '/api/admin/projects/{projectId}/categories',
        pathParams: { projectId },
        body: { name },
      });
      return data;
    },
    async onSuccess() {
      await refetch();
      toast.success(t('v2.toast.success'));
    },
    onError(error) {
      toast.error(error.message);
    },
  });

  return (
    <Combobox>
      <ComboboxTrigger asChild>
        <button
          onClick={(e) => e.stopPropagation()}
          disabled={!perms.includes('issue_update')}
          className={cn(
            { 'opacity-50': !perms.includes('issue_update') },
            'w-fit',
          )}
        >
          {children}
        </button>
      </ComboboxTrigger>
      <ComboboxContent>
        <ComboboxInput
          onClick={(e) => e.stopPropagation()}
          onValueChange={(value) => setInputValue(value)}
          value={inputValue}
        />
        <ComboboxList onClick={(e) => e.stopPropagation()} maxHeight="200px">
          <ComboboxGroup
            heading={
              <span className="text-neutral-tertiary text-base-normal">
                Selected Category
              </span>
            }
          >
            {data?.items
              .filter((v) => category?.id === v.id)
              .map(({ id, name }) => (
                <ComboboxItem
                  key={id}
                  value={name}
                  onSelect={() => {
                    if (category?.id === id) detachCategory({ categoryId: id });
                    if (category?.id !== id) attachCategory({ categoryId: id });
                  }}
                >
                  <span className="flex-1">{name}</span>
                  <span className="text-neutral-tertiary text-small-normal">
                    Remove
                  </span>
                </ComboboxItem>
              ))}
            <InfiniteScrollArea
              fetchNextPage={() => setPage(page + 1)}
              hasNextPage={
                (data?.meta.itemCount ?? 0) < (data?.meta.totalItems ?? 0)
              }
            />
          </ComboboxGroup>
          <ComboboxGroup
            heading={
              <span className="text-neutral-tertiary text-base-normal">
                Category List
              </span>
            }
          >
            {data?.items
              .filter((v) => category?.id !== v.id)
              .map(({ id, name }) => (
                <ComboboxItem
                  key={id}
                  value={name}
                  onSelect={() => {
                    if (category?.id === id) detachCategory({ categoryId: id });
                    if (category?.id !== id) attachCategory({ categoryId: id });
                  }}
                >
                  <span className="flex-1">{name}</span>
                  <CategoryComboboxEditPopover
                    projectId={projectId}
                    cateogry={{ id, name }}
                  />
                </ComboboxItem>
              ))}
            <InfiniteScrollArea
              fetchNextPage={() => setPage(page + 1)}
              hasNextPage={
                (data?.meta.itemCount ?? 0) < (data?.meta.totalItems ?? 0)
              }
            />
          </ComboboxGroup>
          {isLoading && <div className="combobox-item">Loading...</div>}
          {!!inputValue &&
            !data?.items.some((v) => v.name === inputValue) &&
            !isLoading && (
              <div
                className="combobox-item"
                onClick={async () => {
                  const name = inputValue.trim();
                  const data = await createCategory({ name });
                  if (!data) return;
                  setInputValue(name);
                  attachCategory({ categoryId: data.id });
                }}
              >
                <span className="flex-1">{inputValue}</span>
                <span className="text-neutral-tertiary text-small-normal">
                  Create
                </span>
              </div>
            )}
        </ComboboxList>
      </ComboboxContent>
    </Combobox>
  );
};

export default CategoryCombobox;
