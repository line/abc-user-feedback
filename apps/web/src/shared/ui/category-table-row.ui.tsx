/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { useEffect, useMemo, useState } from 'react';
import {
  getCoreRowModel,
  getPaginationRowModel,
  useReactTable,
} from '@tanstack/react-table';
import { useTranslation } from 'next-i18next';

import { Badge, Icon } from '@ufb/react';

import type { Category } from '@/entities/category';
import type { Issue } from '@/entities/issue';
import { useIssueSearch } from '@/entities/issue';
import { getColumnsByCategory } from '@/widgets/issue-table/issue-table-columns';
import IssueDetailSheet from '@/widgets/issue-table/ui/issue-detail-sheet.ui';

import { useOAIQuery, useSort } from '../lib';
import type { SearchQuery } from '../types';
import { cn } from '../utils';
import type { TableFilterOperator } from './table-filter-popover';
import { BasicTable, TablePagination } from './tables';

const DEFAULT_META = {
  currentPage: 1,
  totalPages: 0,
  totalItems: 0,
  itemCount: 0,
  itemsPerPage: 0,
};
interface Props {
  projectId: number;
  category: Category;
  queries: SearchQuery[];
  operator: TableFilterOperator;
}

const CategoryTableRow = (props: Props) => {
  const { projectId, category, queries, operator } = props;

  const { t } = useTranslation();

  const [rows, setRows] = useState<Issue[]>([]);
  const [meta, setMeta] = useState(DEFAULT_META);
  const [isOpen, setIsOpen] = useState(category.id === 0);
  const [openIssueId, setOpenIssueId] = useState<number | null>(null);
  const currentIssue = useMemo(
    () => rows.find((v) => v.id === openIssueId),
    [rows, openIssueId],
  );

  const columns = useMemo(
    () => getColumnsByCategory(t, projectId),
    [t, projectId],
  );

  const table = useReactTable({
    columns,
    data: rows,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
    pageCount: meta.totalPages,
    rowCount: meta.totalItems,
    manualPagination: true,
    getRowId: (row) => String(row.id),
  });

  const { pagination, sorting } = table.getState();
  const sort = useSort(sorting);
  const { data, isLoading } = useIssueSearch(projectId, {
    queries: queries.concat([
      { key: 'categoryId', value: category.id, condition: 'IS' },
    ]),
    page: pagination.pageIndex + 1,
    limit: pagination.pageSize,
    operator,
    sort,
  });

  const { data: issueTracker } = useOAIQuery({
    path: '/api/admin/projects/{projectId}/issue-tracker',
    variables: { projectId },
  });

  useEffect(() => {
    setRows(data?.items ?? []);
    setMeta(data?.meta ?? DEFAULT_META);
  }, [data]);

  return (
    <div>
      <div
        className={cn(
          'bg-neutral-tertiary flex h-12 cursor-pointer items-center rounded-t px-4',
          { 'rounded-b': !isOpen },
        )}
        onClick={() => setIsOpen(!isOpen)}
      >
        <div className="flex gap-2">
          <Icon
            name="RiArrowDownSLine"
            className={cn({ 'rotate-180': !isOpen })}
          />
          <span className={cn({ 'text-neutral-tertiary': category.id === 0 })}>
            {category.name}
          </span>
          {data?.meta.totalItems ?
            <Badge variant="outline" radius="large">
              {data.meta.totalItems}
            </Badge>
          : <></>}
        </div>
      </div>
      {isOpen && (
        <>
          <BasicTable
            table={table}
            disableRound
            onClickRow={(_, row) => setOpenIssueId(row.id)}
            isLoading={isLoading}
            emptyCaption="No issues found"
          />
          <div className="border-neutral-tertiary flex h-12 w-full items-center rounded-b border border-t-0 px-4">
            <TablePagination table={table} disableRowSelect disableLimit />
          </div>
        </>
      )}
      {currentIssue && (
        <IssueDetailSheet
          isOpen={!!currentIssue}
          close={() => setOpenIssueId(null)}
          data={currentIssue}
          projectId={projectId}
          issueTracker={issueTracker?.data}
        />
      )}
    </div>
  );
};

export default CategoryTableRow;
