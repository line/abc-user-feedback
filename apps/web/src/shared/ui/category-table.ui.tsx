/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { useInfiniteQuery } from '@tanstack/react-query';
import { useTranslation } from 'next-i18next';

import { Icon } from '@ufb/react';

import { client } from '../lib';
import type { SearchQuery } from '../types';
import CategoryTableRow from './category-table-row.ui';
import InfiniteScrollArea from './infinite-scroll-area.ui';
import type { TableFilterOperator } from './table-filter-popover';

interface Props {
  projectId: number;
  queries: SearchQuery[];
  operator: TableFilterOperator;
}

const CategoryTable = (props: Props) => {
  const { projectId, queries, operator } = props;
  const { t } = useTranslation();

  const { data, fetchNextPage, hasNextPage, isFetchingNextPage } =
    useInfiniteQuery({
      queryKey: ['/api/admin/projects/{projectId}/categories', projectId],
      queryFn: async ({ pageParam }) => {
        const { data } = await client.get({
          path: '/api/admin/projects/{projectId}/categories',
          pathParams: { projectId, page: pageParam },
        });
        return data;
      },
      getNextPageParam: (lastPage) => {
        if (lastPage.meta.currentPage < lastPage.meta.totalPages) {
          return lastPage.meta.currentPage + 1;
        }
        return undefined;
      },
      initialPageParam: 1,
      initialData: { pageParams: [], pages: [] },
    });

  return (
    <div className="flex flex-col gap-3">
      {[
        { id: 0, name: t('v2.text.no-category') },
        ...data.pages.flatMap((v) => v.items),
      ].map((item) => (
        <CategoryTableRow
          key={item.id}
          category={item}
          projectId={projectId}
          queries={queries}
          operator={operator}
        />
      ))}
      <InfiniteScrollArea
        fetchNextPage={fetchNextPage}
        hasNextPage={hasNextPage}
      />
      {isFetchingNextPage && (
        <div className="flex justify-center">
          <Icon name="RiLoader4Line" className="spinner" />
        </div>
      )}
    </div>
  );
};

export default CategoryTable;
