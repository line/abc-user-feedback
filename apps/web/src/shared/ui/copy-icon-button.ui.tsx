/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { useTranslation } from 'next-i18next';

import { Button, Icon, toast } from '@ufb/react';

interface Props {
  data: string;
}

const CopyIconButton: React.FC<Props> = (props) => {
  const { data } = props;
  const { t } = useTranslation();
  return (
    <Button
      variant="ghost"
      onClick={async (e) => {
        e.stopPropagation();
        try {
          await navigator.clipboard.writeText(data);
          toast(t('v2.toast.copy'), {
            icon: <Icon name="RiCheckboxMultipleFill" />,
          });
        } catch (error) {
          console.error(error);
        }
      }}
    >
      <Icon name="RiLinkM" size={16} className="cursor-pointer" />
    </Button>
  );
};

export default CopyIconButton;
