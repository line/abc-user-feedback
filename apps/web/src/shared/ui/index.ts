/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
export { default as Logo } from './logo.ui';
export { default as LocaleSelectBox } from './locale-select-box.ui';
export { default as LogoWithTitle } from './logo-with-title.ui';

export { default as SectionTemplate } from './section-template.ui';
export { default as MainCard } from './main-card.ui';

export { default as SubMenu } from './sub-menu.ui';
export { default as CreateTemplate } from './create-template.ui';
export { default as CreateInputTemplate } from './create-input-template.ui';
export { default as CreateSectionTemplate } from './create-section-template.ui.tsx';
export { default as DashboardCard } from './dashboard-card';

export { default as ExpandableText } from './expandable-text.ui';
export { default as ImagePreviewButton } from './image-preview-button';
export { default as ShareButton } from './share-button';
export { default as DateRangePicker } from './date-range-picker';
export { default as DescriptionTooltip } from './description-tooltip';
export { default as HelpCardDocs } from './help-card-docs';
export { default as Popper } from './popper.ui';
export { default as ImageSlider } from './image-slider.ui';
export { default as RadioGroup } from './radio-group';

export { default as SmallCard } from './small-card.ui';

export * from './tables';
export * from './charts';
export * from './select-box';
