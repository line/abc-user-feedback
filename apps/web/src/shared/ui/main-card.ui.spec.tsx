/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { faker } from '@faker-js/faker';

import { IconNames } from '@ufb/ui';

import { render } from '@/test-utils';
import MainCard from './main-card.ui';

describe('MainCard', () => {
  test('snapshot', () => {
    const { container } = render(
      <MainCard
        icon={{
          bgColor: '#48DECC',
          iconName: faker.helpers.arrayElement(IconNames),
        }}
        leftContent={{
          title: faker.lorem.word(),
          count: faker.number.int(),
        }}
        rightContent={{
          title: faker.lorem.word(),
          count: faker.number.int(),
        }}
        title={faker.lorem.word()}
        description={faker.lorem.words()}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
