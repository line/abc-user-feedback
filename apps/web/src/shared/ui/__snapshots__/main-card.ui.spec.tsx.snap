// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard snapshot 1`] = `
<div>
  <div
    class="card h-[204px] w-[452px] rounded border p-8"
  >
    <div
      class="mb-10 flex gap-5"
    >
      <div
        class="flex h-10 w-10 items-center justify-center rounded"
        style="background-color: rgb(72, 222, 204);"
      >
        <div
          class="inline-block text-inverse"
          height="20"
          width="20"
        />
      </div>
      <div
        class="flex-1"
      >
        <p
          class="font-16-bold mb-1"
        >
          est
        </p>
        <p
          class="font-12-regular text-secondary line-clamp-1 break-all"
        >
          tutamen arcesso abeo
        </p>
      </div>
    </div>
    <div
      class="flex gap-16"
    >
      <div>
        <p
          class="font-12-regular mb-1"
        >
          speciosus
        </p>
        <p
          class="font-24-bold"
        >
          2,507,328,496,599,040
        </p>
      </div>
      <div>
        <p
          class="font-12-regular mb-1"
        >
          cupiditas
        </p>
        <p
          class="font-24-bold"
        >
          3,823,714,539,929,600
        </p>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;
